src/transaction/TransactionController.ts

```diff
@@ -255,14 +256,15 @@ export class TransactionController extends BaseController<TransactionConfig, Tra
    * @param currentNetworkID - string representing the current network id
```

Comment:
Nit: The additional parameter should be documented here:
```suggestion
   * @param currentNetworkID - string representing the current network id
   * @param currentChainId - The current chain ID
```

Reply:
> I'm not sure what to think of this "backwards compatibility" strategy 🤔 I think it will ensure things are _no worse_ than they are today, but it'll still result in some interactions with the wrong transaction (e.g. if you have two networks with the same network ID but different chain IDs).
> 
> What's worse, this flaw will apply not just to old transactions, but to new ones as well.
> 
> I can approve this anyway (after tests have been added to pass the coverage requirements on CI), since it's less broken than before. But this remains somewhat broken. Maybe this could be improved in the future by removing the "networkID" property for new transactions, and only considering that for older transactions.

@Gudahtt I am 100% with you on all your comments. The end goal is to remove networkID completely. We are doing this incrementally because we still use it in some parts of the app so I thought it would be safer to keep it but always prefer chainId over it.

I have changed some part of the code to be sure that chainId is always preferred, we only fallback to networkID in case chainId is not there. For all new transactions, chainId will always be added, so at least we know that for new transactions, networkID will not be used.

We had a change some time ago where we basically did ` const currentNetworkID = network.state.provider.chainId;` which could cause problems like you mentioned: could result in the wrong set of transactions being returned whenever the chainId was different from the networkId. I now changed to:
` 
const currentChainId= network.state.provider.chainId;
const currentNetworkID= network.state.network;
` 
In this way, chainId and networkID at least can now have different values resolving that problem for now, but yes the objective is still to remove networkID completely in the near future.

Let me know what are your thoughts on this and if it's ok to merge!

Reply:
That sounds good to me! Better to make these changes incrementally if possible